src/gui/editlocallyjob.cpp

```diff
@@ -542,16 +541,112 @@ void EditLocallyJob::openFile()
         return;
     }
 
-    const auto localFilePath = _localFilePath;
+    const auto localFilePathUrl = QUrl::fromLocalFile(_localFilePath);
     // In case the VFS mode is enabled and a file is not yet hydrated, we must call QDesktopServices::openUrl
     // from a separate thread, or, there will be a freeze. To avoid searching for a specific folder and checking
     // if the VFS is enabled - we just always call it from a separate thread.
-    QtConcurrent::run([localFilePath]() {
-        QDesktopServices::openUrl(QUrl::fromLocalFile(localFilePath));
+    QtConcurrent::run([localFilePathUrl, this]() {
+        if(QDesktopServices::openUrl(localFilePathUrl)) {
+            lockFile();
```

Comment:
@claucambra I am not convinced this is the right place to call something that starts network requests. Probably, it is a limitation by Qt (IIRC network calls must be made from the main thread). I have been testing this PR and I can see that files are not always locked remotely.

See which signals are triggered when you put breakpoints in the `void Account::setLockFileState(const QString &serverRelativePath,
                               SyncJournalDb * const journal,
                               const SyncFileItem::LockStatus lockStatus)`.
When locking/unlocking files via context menu - those signals will get triggered. When locking the file via `Edit Locally`, Lock job never finishes. It just starts and the `finished` signal is never called.
If you move this `lockFile()` call outside of the `QtConcurrent::run`, you will have noticed the signals are now getting emitted properly.
I propose you do the locking as a last step in the sequence, and then call ```
QtConcurrent::run([localFilePathUrl, this]() {
     QDesktopServices::openUrl(localFilePathUrl);
});
```
after the LockJob has finished (successfully) or if it never ran because the file was already locked.

So we go:

1. Make the local file up-to-date on disk
2. Lock it if needed and wait till lock job succeeds
3. Call the `openUrl` function after the LockJob has succeeded or if the file was already locked.

Makes sense?

Reply:
Big thanks for the breakdown, I had forgotten that the network requests could only be done on the main thread and this explains a lot of the weirdness around the lockfilejob signals

I have moved the locking procedure onto the main thread and this should now be working correctly 🙂